modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/QuantumToolkit.java

```diff
@@ -633,7 +633,11 @@ void vsyncHint() {
         }
 
         if (!canStartNestedEventLoop()) {
-            throw new IllegalStateException("Cannot enter nested loop during animation or layout processing");
+            if (!Application.GetApplication().canStartNestedEventLoop()) {
+                throw new RuntimeException("Exceeded limit on nested event loops");
```

Comment:
should we tell what the limit is (using the static constant mentioned earlier)?
e.g.
"Too many nested event loops (250)"? or something like that.

Reply:
The number is in the Mac-only Glass code and may change over time so it would be a maintenance burden to repeat it here. But if a developer does see this exception I'm sure their first question would be what the limit is. I can update the comment with something vague (like "over > 240") but I'm hoping that the very, very long Java stack trace attached to this exception will be enough.

Reply:
Unless you are going to track the actual depth, I'd probably leave it as is. When Java throws a StackOverflow it doesn't print the depth.

Reply:
what I meant is the exception should indicate the limit that triggered the exception.
because it will likely be the first question the application developer asks when seeing this exception.

Reply:
The limit is an implementation detail, and should not be elevated to specification. What application developers should be doing if they want to avoid hitting the limit, is call `Platform.canStartNestedEventLoop()`.

Reply:
This is more like the StackOverflow case where the nesting level has gone too deep. It wouldn't hurt tell the application how deep they are (and from that they could infer what the limit is for the platform), but I'm not sure how useful it would be to the application developer. I don't mind one way or the other.

If we do decide to have a platform-independent nesting level, we would specify that limit. I'm still not sure whether we want to do this, but as I mentioned in another comment, there are some advantages to it.